tools/booster/Proxy.hs

```diff
@@ -275,28 +280,35 @@ respondEither mbStatsVar booster kore req = case req of
-                Right (Simplify simplified) -> do
-                    -- to convert back to a term/constraints form,
-                    -- we run a trivial execute request (in booster)
-                    -- We cannot call the booster internaliser without access to the server state
-                    -- Again this should not fail.
-                    let request =
-                            (emptyExecuteRequest simplified.state)
-                                { _module = mbModule
-                                , maxDepth = Just $ Depth 0
-                                }
-                    Log.logInfoNS "proxy" "Making 0-step execute request to convert back to a term/constraints form"
-                    result <- booster $ Execute request
-                    case result of
-                        Right (Execute ExecuteResult{state = finalState}) ->
-                            -- return the state converted by the Booster and logs from simplification.
-                            -- The logs from the 0-step execute request because will be empty, we don't request them
-                            pure (finalState, combineLogs [simplified.logs])
-                        _other -> pure (s, Nothing)
+                Right (Simplify simplified)
+                    | KoreJson.KJBottom _ <- simplified.state.term ->
+                        pure (Left simplified.logs)
+                    | otherwise -> do
+                        -- to convert back to a term/constraints form,
+                        -- we run a trivial execute request (in booster)
+                        -- We cannot call the booster internaliser without access to the server state
+                        -- This call would fail for a \bottom state
+                        let request =
+                                (emptyExecuteRequest simplified.state)
+                                    { _module = mbModule
+                                    , maxDepth = Just $ Depth 0
+                                    }
+                        Log.logInfoNS "proxy" "Making 0-step execute request to convert back to a term/constraints form"
+                        result <- booster $ Execute request
+                        case result of
+                            Right (Execute ExecuteResult{state = finalState}) ->
+                                -- return state converted by Booster and logs from simplification.
+                                -- The 0-step execute result won't have logs (no logs are requested)
+                                pure $ Right (finalState, simplified.logs)
+                            other -> do
+                                Log.logWarnNS "proxy" $
+                                    "Error in pseudo-execute step after simplification: "
+                                        <> either (Text.pack . show) (Text.pack . show . encode) other
+                                pure $ Right (s, Nothing)
```

Comment:
perhaps we should just thread the server state here and do the conversion only? might be worth if we need to do a bigger refactor of the proxy?

Reply:
I agree that this should be done at some point, but at a low priority.

Reply:
Agree. I would just hope that we don't have to do that simplification business at all... but having access to the server state would be cleaner.